src/sql/workbench/contrib/notebook/browser/models/notebookContexts.ts

```diff
@@ -3,113 +3,44 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { nb } from 'azdata';
 
 import { localize } from 'vs/nls';
-import { IDefaultConnection } from 'sql/workbench/contrib/notebook/browser/models/modelInterfaces';
-import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
 import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
-import { IConnectionProfile } from 'sql/platform/connection/common/interfaces';
 import { mssqlProviderName } from 'sql/platform/connection/common/constants';
-import { find } from 'vs/base/common/arrays';
+
 
 export class NotebookContexts {
 
-	public static get DefaultContext(): IDefaultConnection {
-		let defaultConnection: ConnectionProfile = <any>{
+	public static get DefaultContext(): ConnectionProfile {
+		return <any>{
```

Comment:
>any [](start = 10, length = 3)

Why cast to any? Should this be ConnectionProfile?

Reply:
(same below for LocalContext)

---
In reply to: [354967886](https://github.com/microsoft/azuredatastudio/pull/8582#discussion_r354967886) [](ancestors = 354967886)

Reply:
Sigh. These are fake connection profile objects and I really really really don't like it.

Reply:
Tried to fix it but went down a bit of a rabbit hole, so may have to save that for another PR for more cleanup 😢

Reply:
Yeah. I think the approach i would try if I was fixing this would be to have a context be ConnectionProfile | DefaultContextType, that way instead of doing stuff like checking ID you just check the type of the object at runtime and use that to determine what you do. 

---
In reply to: [354972934](https://github.com/microsoft/azuredatastudio/pull/8582#discussion_r354972934) [](ancestors = 354972934)

Reply:
Makes a ton of sense, thanks